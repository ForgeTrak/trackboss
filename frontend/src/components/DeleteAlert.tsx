import React from 'react';
import {
    Modal,
    ModalOverlay,
    ModalContent,
    Button,
    Divider,
    Heading,
    ModalFooter,
    Text,
} from '@chakra-ui/react';

interface alertProps {
  isOpen: boolean,
  onClose: () => void,
  removeMethod: () => void,
}

export default function DeleteAlert(props: alertProps) {
    return (
        <Modal isCentered size="md" isOpen={props.isOpen} onClose={props.onClose}>
            <ModalOverlay />
            <ModalContent>
                <Heading
                    textAlign="center"
                >
                    Are you sure?
                </Heading>
                <Text textAlign="center" fontSize="2xl">You can&apos;t undo this action afterwards.</Text>
                <Divider />
                <ModalFooter>
                    <Button
                        variant="ghost"
                        mr={3}
                        size="lg"
                        onClick={
                            () => {
                                props.onClose();
                            }
                        }
                    >
                        Cancel
                    </Button>
                    <Button
                        color="red"
                        variant="ghost"
                        size="lg"
                        onClick={
                            () => {
                                props.removeMethod();
                                props.onClose();
                            }
                        }
                    >
                        Delete
                    </Button>
                </ModalFooter>
            </ModalContent>
        </Modal>
    );
}
