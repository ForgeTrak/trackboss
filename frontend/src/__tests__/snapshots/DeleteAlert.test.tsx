import React from 'react';
import renderer, { act, ReactTestRenderer } from 'react-test-renderer';
import DeleteAlert from '../../components/DeleteAlert';

function remove() {
    // void function
}

function onClose() {
    // void function
}

// Necessary to mock the chakra-ui modal portal, so that the renderer can target the modal instead of the page
const divWithChildrenMock = (children: any, identifier: any) => <div data-testId={identifier}>{children}</div>;
jest.mock('@chakra-ui/react', () => (
    {
        ...jest.requireActual('@chakra-ui/react'),
        PortalManager: jest.fn(({ children }) => divWithChildrenMock(children, 'portal')),
    }
));

describe('delete alert', () => {
    it('renders correctly', () => {
        let modal: ReactTestRenderer;
        act(() => {
            modal = renderer.create(<DeleteAlert isOpen onClose={onClose} removeMethod={remove} />);
        });

        expect(modal!.toJSON).toMatchSnapshot();
    });
});
