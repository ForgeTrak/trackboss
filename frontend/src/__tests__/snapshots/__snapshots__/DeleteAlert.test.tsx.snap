// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete alert renders correctly 1`] = `[Function]`;
